tgui/packages/tgui-panel/chat/model.js

```diff
@@ -40,6 +39,12 @@ export const createMessage = payload => ({
 export const serializeMessage = message => ({
   type: message.type,
   text: message.text,
+  html: message.html,
   times: message.times,
   createdAt: message.createdAt,
 });
+
+export const isSameMessage = (a, b) => (
+  typeof a.text === 'string' && a.text === b.text
+  || typeof a.html === 'string' && a.html === b.html
```

Comment:
What's the typeof checks here for? Is this to just check if text and html are defined? Why not explicitly check them for null/undefined?

Reply:
Well, it's because `null` and `undefined` are two separate values, and checking for them requires two separate comparisons. Value might also accidentally be `true` or `false`, which we also do not want to compare. Basically, if it's a string, then we assume it is defined. We only expect strings there.

Reply:
That's true. You could do `!a.text` but I'm pretty sure that captures for `"0"` as well.

Reply:
There's also `.isNullOrUndefined()` if you want to be explicit, but doesn't really matter.